cardano-testnet/test/cardano-testnet-test/Cardano/Testnet/Test/Cli/Query.hs

```diff
@@ -78,101 +112,280 @@ hprop_cli_queries = integrationWorkspace "cli-queries" $ \tempAbsBasePath' -> H.
-    -- to stdout
-    stakePoolsOut <- execCli' execConfig [ eraName, "query", "stake-pools" ]
-    H.assertWith stakePoolsOut $ \pools ->
-      length (lines pools) == 3 -- Because, by default, 3 stake pools are created
-    -- Light test of the query's answer, the ids should exist:
-    forM_ (lines stakePoolsOut) $ \stakePoolId -> do
-      execCli' execConfig [ eraName, "query", "pool-state"
-                          , "--stake-pool-id", stakePoolId ]
-    -- to a file
-    let stakePoolsOutFile = work </> "stake-pools-out.json"
-    H.noteM_ $ execCli' execConfig [ eraName, "query", "stake-pools" , "--out-file", stakePoolsOutFile]
-
-  -- stake-distribution
-  do
-    -- to stdout
-    stakeDistrOut <- execCli' execConfig [ eraName, "query", "stake-distribution" ]
-    -- stake addresses with stake
-    let stakeAddresses :: [(Text, Text)] =
-          map
-          ( bimap T.strip T.strip
-            . T.breakOn " " -- separate address and stake
-            . T.strip
-            . fromString )
-          . drop 2 -- drop header
-          . lines
-          $ stakeDistrOut
-    H.assertWith stakeAddresses $ \sa ->
-      -- Because, by default, 3 stake pools are created
-      length sa == 3
-    -- Light test of the query's answer, the ids should exist:
-    forM_ stakeAddresses $ \(stakePoolId, _) -> do
-      execCli' execConfig [ eraName, "query", "pool-state"
-                          , "--stake-pool-id", T.unpack stakePoolId ]
-    -- to a file
-    let stakePoolsOutFile = work </> "stake-distribution-out.json"
-    H.noteM_ $ execCli' execConfig [ eraName, "query", "stake-distribution"
-                                   , "--out-file", stakePoolsOutFile]
-
-  -- gov-state
-  do
-    -- to stdout
-    execCli' execConfig [ eraName, "query", "gov-state" ]
-      >>=
-        (`H.diffVsGoldenFile`
-            "test/cardano-testnet-test/files/golden/queries/govStateOut.json")
-    -- to a file
-    let govStateOutFile = work </> "gov-state-out.json"
-    H.noteM_ $ execCli' execConfig [ eraName, "query", "gov-state", "--out-file", govStateOutFile]
-    H.diffFileVsGoldenFile
-      govStateOutFile
-      "test/cardano-testnet-test/files/golden/queries/govStateOut.json"
-
-  -- drep-state
-  do
-    -- to stdout
-    -- TODO: deserialize to a Haskell value when
-    -- https://github.com/IntersectMBO/cardano-cli/issues/606 is tackled
-    dreps :: Aeson.Value <- H.noteShowM $ execCliStdoutToJson execConfig [ eraName, "query", "drep-state", "--all-dreps"]
-    assertArrayOfSize dreps 3
-    -- to a file
-    let drepStateOutFile = work </> "drep-state-out.json"
-    H.noteM_ $ execCli' execConfig [ eraName, "query", "drep-state", "--all-dreps"
-                                   , "--out-file", drepStateOutFile]
-    _ :: Aeson.Value <- H.readJsonFileOk drepStateOutFile
-    pure ()
-
-  H.success
+  -- If we don't wait, the leadershi-schedule test will say SPO has no stake
+  _ <- waitForEpochs epochStateView (EpochInterval 1)
```

Comment:
Would it make sense to move leadership schedule test case and wait to the end? That should make this test a little bit faster.

Reply:
Probably, but unfortunately changing the order of the cases won't change the order of execution, because the order of execution is generated by template haskell (there is a generated function that calls the case for every constructor). I think it follows the order in which the constructors were defined, so we could reorder those, or make some more complicated machinery that allows specifying order somehow, but I was trying to resist doing any of that